From b738639ea4281f038c09a3ddeef724dd8619454b Mon Sep 17 00:00:00 2001
From: "yang.wang" <yang.wang@deepcomputing.io>
Date: Thu, 5 Sep 2024 16:25:11 +0800
Subject: [PATCH 164/214] ASOC: codecs: ES8326: Init the default mic channel.

Signed-off-by: yang.wang <yang.wang@deepcomputing.io>
---
 sound/soc/codecs/es8326.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/sound/soc/codecs/es8326.c b/sound/soc/codecs/es8326.c
index 58675ad41b5e..749adf8344b3 100755
--- a/sound/soc/codecs/es8326.c
+++ b/sound/soc/codecs/es8326.c
@@ -5,7 +5,7 @@
 //
 // Authors: David Yang <yangxiaohua@everest-semi.com>
 //
-
+#define DEBUG
 #include <linux/clk.h>
 #include <linux/i2c.h>
 #include <linux/interrupt.h>
@@ -996,6 +996,12 @@ static int es8326_probe(struct snd_soc_component *component)
 	dev_dbg(component->dev, "interrupt-clk %x", es8326->interrupt_clk);
 
 	es8326_resume(component);
+
+#ifdef STARFIVE_CONFIG_CODEC_ES8326
+	regmap_write(es8326->regmap, ES8326_ADC_MUTE, 0x0f);
+	regmap_write(es8326->regmap, ES8326_ADC1_SRC, 0x44);
+	regmap_write(es8326->regmap, ES8326_ADC2_SRC, 0x66);
+#endif
 	return 0;
 }
 
-- 
2.54.0

