From 365eb225ec7effa893fb17df8fe5e51532a68ee0 Mon Sep 17 00:00:00 2001
From: Jason Montleon <jmontleo@redhat.com>
Date: Mon, 17 Feb 2025 11:14:30 -0500
Subject: [PATCH 214/214] Remove C++ style comments

---
 include/uapi/linux/jh7110-isp.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/uapi/linux/jh7110-isp.h b/include/uapi/linux/jh7110-isp.h
index 33e0da35c291..c5628e105644 100644
--- a/include/uapi/linux/jh7110-isp.h
+++ b/include/uapi/linux/jh7110-isp.h
@@ -331,11 +331,11 @@ struct jh7110_isp_sc_setting {
 
 struct jh7110_isp_outss_setting {
 	__u8 which;
-	__u16 stride;	// Output Image Stride Register, 8-byte(64bit) granularity.
-	__u8 hsm;		// horizontal scale mode
-	__u32 hsf;		// horizontal scale factor (time 4096)
-	__u8 vsm;		// vertical scale mode
-	__u32 vsf;		// vertical scale factor (time 4096)
+	__u16 stride;	/* Output Image Stride Register, 8-byte(64bit) granularity. */
+	__u8 hsm;		/* horizontal scale mode */
+	__u32 hsf;		/* horizontal scale factor (time 4096) */
+	__u8 vsm;		/* vertical scale mode */
+	__u32 vsf;		/* vertical scale factor (time 4096) */
 };
 
 struct jh7110_isp_sc_buffer {
-- 
2.51.0

